docs/C_API.md

```diff
@@ -0,0 +1,137 @@
+# Clspv - C API wrapper
+
+This folder contains the C API for shared lib version of Clspv.
+
+## clBuildProgram Example Flow
+
+```C++
+#include <cstdio>
+#include <cstdint>
+
+#include "clspv/Compiler.h"
+
+int main() {
+    // CL source example:
+    const char* clSrc = "__kernel void sample_test(__global int *dst, int count) {  \n"
+                        "                                                           \n"
+                        "   int  tid = get_global_id(0);                            \n"
+                        "   for( int i = 0; i < count; i++ )                        \n"
+                        "       dst[i] = tid + i;                                   \n"
+                        "}                                                          \n";
+
+    // Compile
+    size_t outputSpvSize = 0;
+    char* outputSpv = nullptr;
+    char* outputBuildLog = nullptr;
+    ClspvError ret = clspvCompileFromSourcesString(1,
+                                                   NULL,
+                                                   (const char**)&clSrc,
+                                                   "--output-format=spv ",
```

Comment:
This is the default value, thus not really needed

Reply:
Ack